File: 0mq/java/push-pull/Consumer.java
import org.zeromq.ZMQ;
import org.zeromq.ZContext;

public class Consumer
{
    static final int PORT = 5555;

    static void receive_messages(ZMQ.Socket socket)
    {
        int cnt = 0;
        while (true) {
            byte[] raw_message = socket.recv(0);
            cnt++;
            String message = new String(raw_message, ZMQ.CHARSET);
            System.out.println("Received message: " + message);
        }
    }

    public static void main(String[] args) throws Exception
    {
        try (ZContext context = new ZContext()) {
            ZMQ.Socket socket = context.createSocket(ZMQ.PULL);

            String address = "tcp://localhost:" + PORT;
            socket.connect(address);
            System.out.println("Connected to " + address);

            System.out.println("Waiting for message...");

            receive_messages(socket);
        }
    }
}

File: 0mq/python/example23_push_pull_streamer/consumer.py
import time

import zmq
from zmq.decorators import context, socket

CONNECTION_TYPE = zmq.PULL
PORT = 5551


@context()
@socket(CONNECTION_TYPE)
def start_consumer(port, context, socket):
    """Spuštění konzumenta."""

    address = "tcp://localhost:{port}".format(port=port)
    socket.connect(address)
    print("Connected to {a}".format(a=address))

    print("Waiting for message...")
    cnt = 0
    while True:
        message = socket.recv_string()
        cnt += 1
        print("Received message {c} of 100: '{m}'".format(c=cnt, m=message))
        time.sleep(0)


start_consumer(PORT)
